biggus/tests/unit/test_Array.py

```diff
@@ -144,5 +145,93 @@ def test_unhashable(self):
             hash(array)
 
 
+class AssertElementwiseMixin(object):
+    def assertElementwise(self, ag1, ag2):
+        self.assertIs(ag1._array1, ag2._array1)
+        self.assertIs(ag1._array2, ag2._array2)
+        self.assertIs(ag1._numpy_op, ag2._numpy_op)
+        self.assertIs(ag1._ma_op, ag2._ma_op)
+
+
+class Test___add__(unittest.TestCase, AssertElementwiseMixin):
+    def test_other_array(self):
+        a = FakeArray([2, 4])
+        b = FakeArray([2, 4])
+        r = a + b
+        self.assertIsInstance(r, biggus._Elementwise)
+        self.assertElementwise(r, biggus.add(a, b))
+
+    def test_other_no_good(self):
+        a = FakeArray([2, 2])
+        with self.assertRaisesRegexp(TypeError, 'unsupported operand type'):
+            a + None
+
+
+class Test___sub__(unittest.TestCase, AssertElementwiseMixin):
+    def test_other_array(self):
+        a = FakeArray([2, 4])
+        b = FakeArray([2, 4])
+        r = a - b
+        self.assertIsInstance(r, biggus._Elementwise)
+        self.assertElementwise(r, biggus.sub(a, b))
+
+    def test_other_no_good(self):
+        a = FakeArray([2, 2])
+        with self.assertRaisesRegexp(TypeError, 'unsupported operand type'):
+            a - None
+
+
+class Test___mul__(unittest.TestCase, AssertElementwiseMixin):
+    def test_other_array(self):
+        a = FakeArray([2, 4])
+        b = FakeArray([2, 4])
+        r = a * b
+        self.assertIsInstance(r, biggus._Elementwise)
+        self.assertElementwise(r, biggus.multiply(a, b))
+
+    def test_other_no_good(self):
+        a = FakeArray([2, 2])
+        with self.assertRaisesRegexp(TypeError, 'unsupported operand type'):
+            a * None
+
+
+class Test___trudiv__(unittest.TestCase, AssertElementwiseMixin):
+    def test_array_truediv(self):
+        # Div and truediv implement the same interface.
+        self.assertIs(biggus.Array.__truediv__.im_func,
+                      biggus.Array.__div__.im_func)
+
+    def test_other_array(self):
+        a = FakeArray([2, 4])
+        b = FakeArray([2, 4])
+        r = a / b
+        self.assertIsInstance(r, biggus._Elementwise)
+        self.assertElementwise(r, biggus.divide(a, b))
+
+    def test_other_no_good(self):
+        a = FakeArray([2, 2])
+        with self.assertRaisesRegexp(TypeError, 'unsupported operand type'):
+            a / None
+
+
+class Test___pow__(unittest.TestCase, AssertElementwiseMixin):
+    def test_array_truediv(self):
+        # Div and truediv implement the same interface.
+        self.assertIs(biggus.Array.__truediv__.im_func,
+                      biggus.Array.__div__.im_func)
```

Comment:
Probable copy'n'paste issue ... `test_array_truediv` shouldn't be here, it's defined in the previous test class.